jib-core/src/test/java/com/google/cloud/tools/jib/http/FailoverHttpClientTest.java

```diff
@@ -333,6 +333,17 @@ public void testGet_originalRequestHeaderUntouchedWhenClearingHeader() throws IO
         "Basic ZmFrZS11c2VybmFtZTpmYWtlLXNlY3JldA==", request.getHeaders().getAuthorization());
   }
 
+  @Test
+  public void testShutDown() throws IOException {
```

Comment:
Can we add a test to verify that previously shutdown connections do not cause this code to break?

Reply:
Added some tests.